### ISSUE TITLE
Louvain Todos

### ISSUE THREAD

**Issue by mknblch** [neo4j-contrib/neo4j-graph-algorithms#701]
**A)** It looks like the computation of the modularity in each "inner" round is correct but we might have a problem when we compare modularity of different graphs (a new virtual graph built from the community structure and the previous graph). There is logic to adjust the weight accordingly but it might be incorrect. Anyway.. we could change this and resume the algo until either the community count doesnt change or max-iterations is reached. And if max-iterations is set to 1 it would just do a single iteration and behave like the "old" louvain algo. And we could get rid of algo.modularityOptimization (or so) completly.

**B)** Its confusing that louvain returns different results if the concurrency is to low (or 1). This was the intended behaviour since its meant to run in parallel but we can easily chang this by turning random iteration order off if concurrency is 1. 
BTW: A config-flag is not a good idea because it would make the concurrency setting useless.

**Comment by mknblch** [comment]
A. already done
B. #704
